rest/config_database.go

```diff
@@ -184,6 +178,13 @@ func DefaultDbConfig(sc *StartupConfig) *DbConfig {
 		Logging:                          DefaultPerDBLogging(sc.Logging),
 	}
 
+	if useXattrs {
+		dbConfig.AutoImport = base.BoolPtr(base.DefaultAutoImport)
+		if base.IsEnterpriseEdition() {
+			dbConfig.ImportPartitions = base.Uint16Ptr(base.GetDefaultImportPartitions(sc.IsServerless()))
```

Comment:
The problem that this is trying to fix is that if you do `/_config?include_runtime=true` with a database without xattrs is that it will not be roundtripable because of the way the configs merge:

Before this change:

Default
AutoImport = true
ImportPartitions = 16

Non Xattrs database
AutoImport = nil
ImportPartitions = nil

When you merge those configs it becomes:

AutoImport = true
ImportPartitions = 16

If you try to PUT a new database, you'll trip over https://github.com/couchbase/sync_gateway/blob/ee619160735d0109c568def3ffc8641a606acfbc/rest/config.go#L765

The other way I thought to fix this problem was to explicitly the values when we have a DbConfig before we add it to server context

AutoImport = nil :
  AutoImport = false if non xattrs
  AutoImport = true if xattrs
ImportPartitions = nil
  ImportPartitions = 0 if non xattrs
  ImportPartions = 16 if xattrs
  
If I change this, there's actually a difference between `ImportPartitions = 0` and `ImportPartitions = nil` and I'd have to change https://github.com/couchbase/sync_gateway/blob/ee619160735d0109c568def3ffc8641a606acfbc/rest/config.go#L768

Reply:
I agree with Tor's approach here, because it's not valid for those properties to be set when running with xattrs disabled, so the default values shown in include_runtime=true should in fact be nil/omitted.

Reply:
I think `import_docs` should still default to false with xattrs disabled - it's a valid config combination to set by default, but I'm OK with omitting partitions given the error when set.

Reply:
Changed this to false.